oldlog.go

```diff
@@ -43,15 +48,40 @@ var loggerMutex sync.RWMutex
 var loggers = map[string]*logging.Logger{}
 
 // SetupLogging will initialize the logger backend and set the flags.
+// TODO calling this in `init` pushes all configuration to env variables
+// - move it out of `init`? then we need to change all the code (js-ipfs, go-ipfs) to call this explicitly
+// - have it look for a config file? need to define what that is
 func SetupLogging() {
 
+	// colorful or plain
 	lfmt := LogFormats[os.Getenv(envLoggingFmt)]
 	if lfmt == "" {
 		lfmt = LogFormats[defaultLogFormat]
 	}
 
-	backend := logging.NewLogBackend(colorable.NewColorableStderr(), "", 0)
-	logging.SetBackend(backend)
+	// check if we log to a file, or syslog, building a list of log backends
+	var lgbe []logging.Backend
+	if logfp := os.Getenv(envLoggingFile); len(logfp) > 0 {
+		f, err := os.Create(logfp)
+		if err != nil {
+			fmt.Printf("ERROR go-log: %s: failed to set logging file backend\n", err)
+		}
+		lgbe = append(lgbe, logging.NewLogBackend(f, "", 0))
```

Comment:
if the above errors, we don't want to add this backend. it will be to a bad file

Reply:
doh! thanks